library/Opus/Doi/DataCiteXmlGenerator.php

```diff
@@ -99,6 +99,22 @@ public function getXml($doc)
         $modelXml = $this->getModelXml($doc);
         $log->debug('OPUS-XML: ' . $modelXml->saveXML());
 
+        $filenodes = $modelXml->getElementsByTagName('File');
+
+        // Iterating over DOMNodeList is only save for readonly-operations -> create separate list
+        $filenodesList = array();
+        foreach ($filenodes as $filenode) {
```

Comment:
Gibt es einen Weg die List (eigentlich ein Array) ohne Schleife zu kopieren? Welchen Typ hat `$filenodes`?

Reply:
Das ist eine DOMNodeList. Laut Kommentaren aus anderen Code-Teilen, ist es schlecht darüber zu iterieren, wenn man nocht nur readonly Operationen hat. In unserem Fall ist das aber nicht der Fall, weshalb wir ein Array benötigen. Für diese Umformung fällt mir keine bessere Alternative ein.